File: lokerpc/codegen/testdata/enum.json.go
package payment

import (
	"context"

	"github.com/LOKE/pkg/lokerpc"
)

type Currency string

const (
	CurrencyAUD Currency = "AUD"
	CurrencyNZD Currency = "NZD"
	CurrencyUSD Currency = "USD"
	CurrencyGBP Currency = "GBP"
	CurrencySGD Currency = "SGD"
)

type PaymentState string

const (
	PaymentStateCreated    PaymentState = "created"
	PaymentStateAuthorized PaymentState = "authorized"
	PaymentStateSubmitted  PaymentState = "submitted"
	PaymentStateSettled    PaymentState = "settled"
	PaymentStateFailed     PaymentState = "failed"
	PaymentStateCanceled   PaymentState = "canceled"
	PaymentStateRefunded   PaymentState = "refunded"
	PaymentStateCompleted  PaymentState = "completed"
)

type GetStateRequest struct {
	Amount   int32    `json:"amount"`
	Currency Currency `json:"currency"`
}

type PaymentService interface {
	GetState(context.Context, GetStateRequest) (*PaymentState, error)
}

type PaymentRPCClient struct {
	lokerpc.Client
}

func (c PaymentRPCClient) GetState(ctx context.Context, req GetStateRequest) (*PaymentState, error) {
	var res PaymentState
	err := c.DoRequest(ctx, "getState", req, &res)
	if err != nil {
		return nil, err
	}
	return &res, nil
}

File: lokerpc/codegen/testdata/enum.json.ts
import { RPCContextClient } from "@loke/http-rpc-client";
import { Context } from "@loke/context";

export type Currency = "AUD" | "NZD" | "USD" | "GBP" | "SGD";

export type PaymentState = "created" | "authorized" | "submitted" | "settled" | "failed" | "canceled" | "refunded" | "completed";

export type GetStateRequest = {
  amount: number;
  currency: Currency;
};

/**
 * Payment service
 */
export class PaymentService extends RPCContextClient {
  constructor(baseUrl: string) {
    super(baseUrl, "payment")
  }
  /**
   * Get payment state
   */
  getState(ctx: Context, req: GetStateRequest): Promise<PaymentState> {
    return this.request(ctx, "getState", req);
  }
}
